auto-pos-creator/src/multi_contract_interactions/create_pos_endpoints.rs

```diff
@@ -100,6 +103,38 @@ pub trait CreatePosEndpointsModule:
         self.create_pos_common(args)
     }
 
+    #[payable("*")]
+    #[endpoint(createPosFromLp)]
+    fn create_pos_from_lp(&self, steps: StepsToPerform) -> EsdtTokenPayment {
+        require!(
+            !matches!(steps, StepsToPerform::AddLiquidity),
```

Comment:
Why not simply check
`require!(steps != StepsToPerform::AddLiquidity, "Invalid step");`

Reply:
I don't have Eq/PartialEq implemented for the enum :)